src/test/java/mate/academy/dao/RoleDaoTest.java

```diff
@@ -0,0 +1,76 @@
+package mate.academy.dao;
+
+import static mate.academy.model.Role.RoleName.ADMIN;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Optional;
+import mate.academy.AbstractTest;
+import mate.academy.dao.impl.RoleDaoImpl;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RoleDaoTest extends AbstractTest {
+    private RoleDao roleDao;
+    private Role role;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[]{Role.class};
+    }
+
+    @BeforeEach
+    void setUp() {
+        roleDao = new RoleDaoImpl(getSessionFactory());
+        role = new Role();
+        role.setRoleName(ADMIN);
+    }
+
+    @Test
+    void save_Ok() {
+        Role actual = roleDao.save(role);
+        assertNotNull(actual);
+        assertEquals(1L, actual.getId());
+        assertEquals(ADMIN, actual.getRoleName());
+    }
+
+    @Test
+    void save_RoleIsNull_NotOk() {
+        assertThrows(DataProcessingException.class, () -> {
+            roleDao.save(null);
+        });
+    }
+
+    @Test
+    void getRoleByName_Ok() {
+        Role admin = new Role(ADMIN);
+        Session session = getSessionFactory().openSession();
+        Transaction transaction = session.beginTransaction();
+        session.save(admin);
+        transaction.commit();
+        session.close();

```

Comment:
Please use related dao methods in both dao tests classes.
I understand that you wanted to isolate get methods from effect of possible failure on save. But unit tests should be as simple as possible. If we have problems with save in test that designed for checking get logic then we will see it during debug in a minute and proceed with bugfixing in code. But if we will get fail on some big logic hardcoded within test then we will need to spend much more additional time to understand the cause of a problem